src/components/MultiCombobox/MenuItemGroups.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { GetItemPropsOptions } from 'downshift';
+import Text from '../Text';
+import Box from '../Box';
+import MenuItem from '../utils/MenuItem';
+import { typedMemo } from '../../utils/helpers';
+import groupBy from 'lodash/groupBy';
+
+interface MenuItemGroupProps<T> {
+  items: T[];
+  disableItem: (item: T) => boolean;
+  getItemProps: (options: GetItemPropsOptions<T>) => T;
+  itemToString: (item: T) => string;
+  itemToGroup: (item: T) => string;
```

Comment:
Shouldn't we have a `null` option in order to force items to not belong in a group?

```suggestion
  itemToGroup: (item: T) => string | null;
```